pkg/sql/catalog/typedesc/table_implicit_record_type.go

```diff
@@ -232,7 +232,7 @@ func (v TableImplicitRecordType) TypeDesc() *descpb.TypeDescriptor {
 func (v TableImplicitRecordType) HydrateTypeInfoWithName(
 	ctx context.Context, typ *types.T, name *tree.TypeName, res catalog.TypeDescriptorResolver,
 ) error {
-	if typ.IsHydrated() {
+	if typ.IsHydrated() && typ.TypeMeta.Version == uint32(v.desc.GetVersion()) {
```

Comment:
This and the changes in the file below are what fix the newly-added test case in the `enums` logic test suite.